src/main/java/ploiu/ui/MainFrame.java

```diff
@@ -57,6 +61,20 @@ public class MainFrame extends AnchorPane {
         folderClient.getFolder(id).ifPresent(this::loadFolder);
         return true;
     };
+    // update folder
+    private final EventReceiver<FolderApi> updateFolderEvents = event -> {
+        var folder = event.get();
+        // I really need to make the api return 0 instead of null for the root folder...
+        var parentId = folder.parentId() == 0 ? null : folder.parentId();
+        try {
+            var updated = folderClient.updateFolder(new FolderRequest(Optional.of(folder.id()), Optional.ofNullable(parentId), folder.path()));
+            // need to redraw the current folder
+            loadFolder(currentFolder);
+            return true;
+        } catch (BadFolderRequestException | BadFolderResponseException e) {
+            return false;
+        }
+    };
```

Comment:
this can probably be consolidated down into one single folder handler...might make the code messy though